synapse/lib/ast.py

```diff
@@ -3958,55 +4202,86 @@ async def walkNodeEdges(self, runt, node, verb=None):
             if walknode is not None:
                 yield walknode
 
-    async def run(self, runt, genr):
+    def buildfilter(self, runt, destforms, cmpr):
 
-        cmpr = None
-        if len(self.kids) == 4:
-            cmpr = await self.kids[2].compute(runt, None)
+        if not isinstance(destforms, (tuple, list)):
+            destforms = (destforms,)
 
-        async def destfilt(destforms, node, path):
+        if '*' in destforms:
+            if cmpr is not None:
+                mesg = 'Wild card walk operations do not support comparison.'
+                raise self.addExcInfo(s_exc.StormRuntimeError(mesg=mesg))
 
-            if not isinstance(destforms, (tuple, list)):
-                destforms = (destforms, )
+            return False
 
-            for destform in destforms:
+        forms = set()
+        formprops = collections.defaultdict(dict)
 
-                if destform == '*':
-                    if cmpr is not None:
-                        mesg = 'Wild card walk operations do not support comparison.'
-                        raise self.addExcInfo(s_exc.StormRuntimeError(mesg=mesg))
-                    return True
+        for destform in destforms:
+            prop = runt.model.prop(destform)
+            if prop is not None:
+                if prop.isform:
+                    forms.add(destform)
+                else:
+                    formprops[prop.form.name][prop.name] = prop
+                continue
 
-                prop = runt.model.prop(destform)
-                if prop is None:
-                    mesg = f'walk operation expects dest to be a prop got: {destform!r}'
-                    raise self.addExcInfo(s_exc.StormRuntimeError(mesg=mesg))
+            if (formlist := runt.model.formsbyiface.get(destform)) is not None:
```

Comment:
Would there be any benefit to utilitizing the formsbyiface / propsbyiface / endswith * logic into a function that returns a list of form names?

Reply:
added some helpers for this